server/server.go

```diff
@@ -0,0 +1,102 @@
+package server
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/rs/zerolog"
+	"github.com/umee-network/osmosis-api/config"
+)
+
+type Server struct {
+	http     http.Server
+	upgrader websocket.Upgrader
+	logger   zerolog.Logger
+	cfg      config.Server
+}
+
+// SocketHandler upgrades our server to use websockets, and registers any msg
+// handlers. Currently only supports "Ping" requests.
+func (s *Server) SocketHandler(w http.ResponseWriter, r *http.Request) {
+	conn, err := s.upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		log.Print("Error during connection upgrade:", err)
+		return
+	}
+	defer conn.Close()
+
+	for {
+		messageType, message, err := conn.ReadMessage()
+		if err != nil {
+			break
+		}
+
+		if bytes.Equal(message, []byte("Ping")) {
+			err = conn.WriteMessage(messageType, []byte("Pong"))
+			if err != nil {
+				break
+			}
+		} else {
+			err = conn.WriteMessage(messageType, []byte("Invalid Request"))
+			if err != nil {
+				break
+			}
+		}
```

Comment:
```suggestion
		if bytes.Equal(message, []byte("Ping")) {
			err = conn.WriteMessage(messageType, []byte("Pong"))
			if err != nil {
				break
			}
			continue
		}
		
		err = conn.WriteMessage(messageType, []byte("Invalid Request"))
		if err != nil {
			break
		}
```
The only suggestion is to try to keep stuffs more in the right